src/Database/Commands/GetStatisticsCommand.ts

```diff
@@ -16,10 +17,12 @@ export class GetStatisticsCommand extends RavenCommand {
   }
 
   public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | void {
-    const responseBody: IResponseBody = response.body;
+    const result: IRavenResponse = <IRavenResponse>super.setResponse(response);
 
-    if (responseBody && StatusCodes.isOk(response.statusCode)) {
-      return responseBody;
+    if (response.body) {
+      return result;
     }
+
+    throw new ErrorResponseException('Invalid server response');
```

Comment:
yes it's possible in future:
- add those props to basic RavenException
- in auto-thrown exceptions those props will be auto-substituted from json response
- instead of manulally throw new ... we will use ExceptionThrower.throwWith(message, type, response)